synapse/storage/push_rule.py

```diff
@@ -184,11 +185,23 @@ def _bulk_get_push_rules_for_room(self, room_id, state_group, current_state_ids,
             if uid in local_users_in_room:
                 user_ids.add(uid)
 
+        forgotten = yield self.who_forgot_in_room(
+            event.room_id, on_invalidate=cache_context.invalidate,
+        )
```

Comment:
I think you are safe to ignore the forgotten stuff if we are only looking for rooms that the user is joined/invited to. I think you can only forget a room that you have left?